### PR TITLE
Refactoring for testing NodePtpDevice api

### PR DESCRIPTION
**Pull request by edcdavid**
Refactor tests checking the output of the NodePtpDevice API ( returning the list of interfaces usable by the ptp-operator in the nodes running the ptp daemonset).
The new test creates a list of non-virtual interfaces supporting PTP and compares it with the list of interfaces returned by the ptpnode API. If the two list matches exactly per ptp pod, the test succeeds.

### CODE REVIEW

File: test/conformance/ptp/ptp.go [hunk 1]
```diff
@@ -184,25 +186,19 @@ var _ = Describe("[ptp]", func() {
...
-					for _, inter := range ptpNotSupportedInt {
-						Expect(ptpDiscoveredInterfaces).ToNot(ContainElement(inter), "The interfaces discovered incorrectly. PTP non supported Interfaces in list")
+					sort.Strings(ptpSupportedInt)
+					sort.Strings(ptpDiscoveredInterfaces)
+					logrus.Infof("Interfaces supporting ptp for node        %s: %v", ptpRunningPods[podIndex].Spec.NodeName, ptpSupportedInt)
+					logrus.Infof("Interfaces discovered by ptp API for node %s: %v", ptpRunningPods[podIndex].Spec.NodeName, ptpDiscoveredInterfaces)
+
+					// The discovered PTP interfaces should match exactly the list of interfaces calculated by test
+					for index := range ptpSupportedInt {
+						Expect(ptpSupportedInt[index]).To(Equal(ptpDiscoveredInterfaces[index]))
```
Comment (greyerof) [inline test/conformance/ptp/ptp.go:201]: Shouldn't you also check that `len(ptpDiscoveredInterfaces) == len(ptpSupportedInt)`?

File: test/conformance/ptp/ptp.go [hunk 2]
```diff
@@ -1428,7 +1396,13 @@ func BasicClockSyncCheck(fullConfig testconfig.TestConfig, ptpConfig *ptpv1.PtpC
 	}
 
 	label, err := metrics.GetLabel(ptpConfig)
+	if err != nil {
+		logrus.Debugf("could not get label because of err: %s", err)
```
Comment (greyerof) [inline test/conformance/ptp/ptp.go:1400]: Why not logrus.Errorf(), or, at least, logrus.Warnf()?

File: test/conformance/ptp/ptp.go [hunk 3]
```diff
@@ -1428,7 +1396,13 @@ func BasicClockSyncCheck(fullConfig testconfig.TestConfig, ptpConfig *ptpv1.PtpC
...
+	}
 	nodeName, err := metrics.GetFirstNode(ptpConfig)
+	if err != nil {
+		logrus.Debugf("could not get nodeName because of err: %s", err)
```
Comment (greyerof) [inline test/conformance/ptp/ptp.go:1404]: Same here.

File: test/conformance/ptp/ptp.go (hunk 2 above)
Comment (edcdavid) [reply test/conformance/ptp/ptp.go:1400]: it is ok to not have a label or a node but not both. The check is done later downstream in getClockIDForeign

File: test/conformance/ptp/ptp.go (hunk 3 above)
Comment (edcdavid) [reply test/conformance/ptp/ptp.go:1404]: it is ok to not have a label or a node but not both. The check is done later downstream in getClockIDForeign

File: test/utils/testconfig/testconfig.go [hunk 4]
```diff
@@ -287,25 +287,26 @@ func CreatePtpConfigurations() {
...
 	GetDesiredConfig(true)
 	// in multi node configuration create ptp configs
+
+	// Initialize l2 library
+	l2lib.GlobalL2DiscoveryConfig.SetL2Client(client.Client, client.Client.Config)
+
+	// Collect L2 info
+	config, err := l2lib.GlobalL2DiscoveryConfig.GetL2DiscoveryConfig(true)
+	if err != nil {
+		return
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply test/utils/testconfig/testconfig.go:298]: fixed

File: test/conformance/ptp/ptp.go (hunk 1 above)
Comment (edcdavid) [reply test/conformance/ptp/ptp.go]: fixed

**Comment by edcdavid** [comment]
ci/prow/operator-e2e

File: go.mod [hunk 5]
```diff
@@ -14,7 +14,8 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.9.0
 	github.com/test-network-function/graphsolver-lib v0.0.0-20220923161551-922faf8697be
```
Comment (aneeshkp) [inline go.mod:16]: This should be upgraded to
v0.0.0-20220926190927-5f72a82c96d4 => v0.0.0-20220929081721-4e9a42510716
or else it error's out
ptpInterface.IfPci.Description undefined

Comment (edcdavid) [reply go.mod]: fixed

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**